### PR TITLE
DDF-04507 Refactored filter-builder to use backbone model / collection

### PR DESCRIPTION
**Pull request by Schachte**
#### What does this PR do?
This is the first part of a refactor with filter-builder to rip out functions from the views and migrate them to models and collections. This will make the migration to React much easier when listening to changes on the model.

This greatly cleans up the flow of how filters get rehydrated within the view and how the data is serialized/deserialized within Intrigue.
#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining 
@djblue 
@andrewkfiedler 
@middlets719 
@gjvera 
@zta6 

#### How should this be tested?

Create queries from advanced builder and ensure they perform as normal.

- Nested groupings, change the operator, add filters, save them and make sure they reload.

#### Any background context you want to provide?
Overall pre-improvement before migrating to react.

#### What are the relevant tickets?
For GH Issues:
Fixes: #4507

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter-builder/filter-builder.view.js [hunk 1]
```diff
@@ -232,14 +167,7 @@ module.exports = Marionette.LayoutView.extend({
     }
   },
   deserialize: function(cql) {
```
Comment (djblue) [inline ui/packages/catalog-ui-search/src/main/webapp/component/filter-builder/filter-builder.view.js]: I don't think this method is needed any longer.

**Comment by Schachte** [comment]
build now

**Comment by Schachte** [comment]
build now

**Comment by middlets719** [comment]
Hero Successful 🎉 
Injected multiple test files.
Perform searches on various attributes and got expected results
Saved Queries and reloaded queries with same results
Verified query preservation for geo searches
Created a list and uploaded data to it
Verified UI searches work in Chrome and Firefox
Verified filterTree is not a search option in the attribute list

**Pull request merged**